octoprint_Spoolman/modules/PrinterHandler.py

```diff
@@ -101,3 +104,73 @@ def commitSpoolUsage(self):
                     'extrusionLength': toolExtrusionLength,
                 }
             )
+
+    def save_Printing_Status_Change(self):
+        data_save = {}
+        peek_stats_helpers = self.lastPrintOdometerLoad.send(False)
+        current_extrusion_stats = copy.deepcopy(peek_stats_helpers['get_current_extrusion_stats']())
+
+        selectedSpoolIds = self._settings.get([SettingsKeys.SELECTED_SPOOL_IDS])
+
+        for toolIdx, toolExtrusionLength in enumerate(current_extrusion_stats['extrusionAmount']):
+            try:
+                selectedSpool = selectedSpoolIds[str(toolIdx)]
+            except:
+                self._logger.info("Extruder '%s', spool id: none", toolIdx)
+
+            if not selectedSpool or selectedSpool.get('spoolId', None) == None:
+                continue
+
+            selectedSpoolId = selectedSpool['spoolId']
+            
+            data_save[str(toolIdx)] = {
+                'toolIdx': toolIdx,
+                'spoolId': selectedSpoolId,
+                'extrusionLength': toolExtrusionLength
+            }
+            
+            self.triggerPluginEvent(
```

Comment:
Triggering this event conceptually does not belong to `save_Printing_Status_Change` function, as it has nothing to do with the backup functionality. This should either be handled in a separate function, or the current function's name should be changed to better reflect what's going on.